### PR TITLE
feat(google_service_account): Add validation for account_id

### PR DESCRIPTION
**Pull request by jai**
## Pull Request Submission Checklist

Please confirm that you have done the following before requesting reviews:

- [x] I have confirmed that the PR type is appropriate for the change I am making according to the [Honest Pull Request and Commit Message Naming Conventions](https://www.notion.so/honestbank/Pull-Request-and-Commit-Message-Naming-Conventions-bd97f2cbb34c4c73b1ff3a3e384b850c).
- [x] I have typed an adequate description that explains **why** I am making this change.
- [x] I have installed and run standard pre-commit hooks that lints and validates my code.

### Description

Google Cloud Platform Service Accounts have restrictions on the account IDs and lengths ([source](https://cloud.google.com/iam/docs/service-accounts-create)).
This PR adds the corresponding validation to the module to prevent apply-time failures.

- - -
This change is [Reviewable](https://reviewable.io/reviews/honestbank/terraform-gcp-iam/26)

### CODE REVIEW

**Comment by jai** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #26** <a href="https://app.graphite.dev/github/pr/honestbank/terraform-gcp-iam/26?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/honestbank/terraform-gcp-iam/26?utm_source=stack-comment).

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -23,3 +16,10 @@ repos:
         exclude: (test/|examples/)
       - id: terraform_checkov
         exclude: (test/|examples/)
+  - repo: https://github.com/pre-commit/pre-commit-hooks
```
Comment (jai) [inline .pre-commit-config.yaml:19]: Changing this until I fix pre-commit in the main repo

Comment (jai) [reply .pre-commit-config.yaml:19]: Addressed in https://github.com/honestbank/.github/pull/259

File: modules/google_service_account/inputs.tf [hunk 2]
```diff
@@ -1,6 +1,10 @@
 variable "account_id" {
   description = "The account id that is used to generate the service account email address and a stable unique id. It is unique within a project, must be 6-30 characters long, and match the regular expression [a-z]([-a-z0-9]*[a-z0-9]) to comply with RFC1035. Changing this forces a new service account to be created."
   type        = string
+  validation {
+    condition     = length(var.account_id) >= 6 && length(var.account_id) <= 30 && regex("[a-z]([-a-z0-9]*[a-z0-9])", var.account_id) == var.account_id
```
Comment (olli-honest) [inline modules/google_service_account/inputs.tf:5]: [Error message](https://app.terraform.io/app/honestbank/workspaces/iac-iam-test/runs/run-BPAThMmyC9rBSZFP):
```
Error: "account_id" ("terraform-ml-infrastructure-test-bdf8d5a5") doesn't match regexp "^[a-z](?:[-a-z0-9]{4,28}[a-z0-9])$"
```
Should we use the same regex?

Comment (jai) [reply modules/google_service_account/inputs.tf]: Thanks, updated

**Review by poom — APPROVED** [review]

**Review by olli-honest — APPROVED** [review]

**Review by MXfive — APPROVED** [review]

**Pull request merged**